@xen-orchestra/lite/src/stores/xen-api/network.store.ts

```diff
@@ -1,9 +1,105 @@
+import type { Status } from '@/components/pool/network/PoolNetworksPifStatus.vue'
+import type { XenApiNetwork, XenApiPif } from '@/libs/xen-api/xen-api.types'
 import { createXapiStoreConfig } from '@/stores/xen-api/create-xapi-store-config'
+import { useHostStore } from '@/stores/xen-api/host.store'
+import { usePifStore } from '@/stores/xen-api/pif.store'
+import { usePoolStore } from '@/stores/xen-api/pool.store'
 import { createSubscribableStoreContext } from '@core/utils/create-subscribable-store-context.util'
+import { sortByNameLabel } from '@core/utils/sort-by-name-label.util'
 import { defineStore } from 'pinia'
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
 
 export const useNetworkStore = defineStore('xen-api-network', () => {
-  const config = createXapiStoreConfig('network')
+  const { t } = useI18n()
 
-  return createSubscribableStoreContext(config, {})
+  const deps = {
+    poolStore: usePoolStore(),
+    hostStore: useHostStore(),
+    pifStore: usePifStore(),
+  }
+
+  const { context: baseContext, ...configRest } = createXapiStoreConfig('network', {
+    sortBy: sortByNameLabel,
+  })
+
+  const poolContext = deps.poolStore.getContext()
+  const hostContext = deps.hostStore.getContext()
+  const pifContext = deps.pifStore.getContext()
+
+  const PIFsByNetwork = computed(() => {
+    const PIFsByNetworkMap = new Map<string, XenApiPif[]>()
+
+    const poolMasterRef = poolContext.pool.value?.master
+
+    const currentHostRef = hostContext.records.value.find(host => host.$ref === poolMasterRef)
+    if (currentHostRef) {
+      pifContext.records.value
+        .filter(pif => pif.host === poolMasterRef)
+        .forEach(pif => {
+          if (!PIFsByNetworkMap.has(pif.network)) {
+            PIFsByNetworkMap.set(pif.network, [])
+          }
+          PIFsByNetworkMap.get(pif.network)?.push(pif)
+        })
+    }
+    return PIFsByNetworkMap
+  })
+
+  const networksWithVLANs = computed(() => {
+    const networksInfoMap = new Map<
+      string,
+      {
+        network: XenApiNetwork
+        vlan: string
+        status: Status
+      }
```

Comment:
Redundant typing should be extracted to a type and be reused.

```suggestion
      {
        network: XenApiNetwork
        vlan: string
        status: Status
      }
```